src/daft-connect/src/display.rs

```diff
@@ -0,0 +1,355 @@
+
+    let dtype_str = type_to_string(dtype);
+    writeln!(
+        w,
+        "{indent}{field_name}: {dtype_str} (nullable = {NULLABLE})"
+    )?;
+
+    if let DataType::Struct(fields) = dtype {
+        for field in fields {
+            write_field(w, &field.name, &field.dtype, level + 1)?;
+        }
+    }
+
+    Ok(())
+}
+
+// This helper creates indentation of the form:
+// level=1: " |-- "
+// level=2: " |    |-- "
+// and so forth.
+fn make_indent(level: usize) -> String {
+    if level == 0 {
+        // If top-level (i.e., a bare field not in a schema), just return empty.
+        String::new()
+    } else if level == 1 {
+        " |-- ".to_string()
+    } else {
+        format!(" |{}-- ", "    |".repeat(level - 1))
+    }
+}
+
+fn type_to_string(dtype: &DataType) -> String {
+    match dtype {
+        DataType::Null => "null".to_string(),
+        DataType::Boolean => "boolean".to_string(),
+        DataType::Int8 => "byte".to_string(),
+        DataType::Int16 => "short".to_string(),
+        DataType::Int32 => "integer".to_string(),
+        DataType::Int64 => "long".to_string(),
+        DataType::Float32 => "float".to_string(),
+        DataType::Float64 => "double".to_string(),
+        DataType::Decimal128(precision, scale) => format!("decimal({precision},{scale})"),
+        DataType::Timestamp(_, _) => "timestamp".to_string(),
+        DataType::Date => "date".to_string(),
+        DataType::Time(_) => "time".to_string(),
+        DataType::Duration(_) => "duration".to_string(),
+        DataType::Interval => "interval".to_string(),
+        DataType::Binary => "binary".to_string(),
+        DataType::FixedSizeBinary(_) => "arrow.fixed_size_binary".to_string(),
+        DataType::Utf8 => "string".to_string(),
+        DataType::FixedSizeList(_, _) => "arrow.fixed_size_list".to_string(),
+        DataType::List(_) => "arrow.list".to_string(),
+        DataType::Struct(_) => "struct".to_string(),
+        DataType::Map { .. } => "map".to_string(),
+        DataType::Extension(_, _, _) => "daft.extension".to_string(),
+        DataType::Embedding(_, _) => "daft.embedding".to_string(),
+        DataType::Image(_) => "daft.image".to_string(),
+        DataType::FixedShapeImage(_, _, _) => "daft.fixed_shape_image".to_string(),
+        DataType::Tensor(_) => "daft.tensor".to_string(),
+        DataType::FixedShapeTensor(_, _) => "daft.fixed_shape_tensor".to_string(),
+        DataType::SparseTensor(_) => "daft.sparse_tensor".to_string(),
+        DataType::FixedShapeSparseTensor(_, _) => "daft.fixed_shape_sparse_tensor".to_string(),
+        #[cfg(feature = "python")]
+        DataType::Python => "daft.python".to_string(),
+        DataType::Unknown => "unknown".to_string(),
+        DataType::UInt8 => "arrow.ubyte".to_string(),
+        DataType::UInt16 => "arrow.ushort".to_string(),
+        DataType::UInt32 => "arrow.uint".to_string(),
+        DataType::UInt64 => "arrow.ulong".to_string(),
```

Comment:
Sorry if I was unclear in my previous comment, but this is still not right.

arrow types should just be called what they are
```rs
        DataType::UInt8 => "arrow.uint8".to_string(),
        DataType::UInt16 => "arrow.uint16".to_string(),
        DataType::UInt32 => "arrow.uint32".to_string(),
        DataType::UInt64 => "arrow.uint64".to_string(),
```

and nested datatypes should match how spark does them
for example, lists have the inner rendered as "element"

```py
data = [{"a": [1,2,3], "b": "hello"}]
spark.createDataFrame(data).printSchema()
```

We'll also want to capture the parameters on them such as `FixedSizeList(Int64, 1)`

```
root
 |-- a: arrow.fixed_size_list (size = 1, nullable = true)
 |    |-- element: long (containsNull = true)

```

or on `Image(ImageMode::RGB)`
```
root
 |-- a: daft.image (mode = RGB, nullable = true)
```


and for structs `Struct{ints: i64, strings: utf8}`

```
root
 |-- struct: struct (nullable = true)
 |    |-- ints: integer (nullable = true)
 |    |-- strings: string (nullable = true)
```